### PR TITLE
Create question service

### PR DESCRIPTION
**Pull request by huizhuansam**
Summary:

Created a Node + Express server and MongoDB database. Wrote Dockerfile and docker-compose. Added API endpoints for C, R functions. Added functionality such as full-text query on title and description of questions. Added synthetic data for local testing.

### CODE REVIEW

File: services/question-service/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - mongodb
+    environment:
+      - MONGO_URI=mongodb://mongodb:27017/question-service-api
+  mongodb:
+    image: mongo:6.0
+    volumes:
+      - ./data:/data/db
+    ports:
+      - "27017:27017"
+
```
Comment (Acerizm) [inline services/question-service/docker-compose.yml:25]: Please bring/appened over the whole docker-compose configs to the main docker-compose.yml file at `Server-Configs` folder

**Review by Acerizm — DISMISSED** [review]
LGTM! I would suggest bringing over the whole docker-compose configs that you have done to the main docker-compose.yml file at `Server-Configs` folder so that we can easily deploy all containers through a single file

File: services/question-service/docker-compose.yml (hunk 1 above)
Comment (huizhuansam) [reply services/question-service/docker-compose.yml:25]: @Acerizm I will open another pull request for this.

**Review by Acerizm — APPROVED** [review]
LGTM!

**Pull request merged**